x-pack/plugins/ml/public/application/model_management/deployment_params_mapper.test.ts

```diff
@@ -0,0 +1,536 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { MlTrainedModelAssignmentTaskParametersAdaptive } from './deployment_params_mapper';
+import { DeploymentParamsMapper } from './deployment_params_mapper';
+import type { CloudInfo } from '../services/ml_server_info';
+import type { MlServerLimits } from '../../../common/types/ml_server_info';
+
+describe('DeploymentParamsMapper', () => {
+  const modelId = 'test-model';
+
+  const mlServerLimits: MlServerLimits = {
+    max_single_ml_node_processors: 10,
+    total_ml_processors: 10,
+  };
+
+  const cloudInfo = {
+    isMlAutoscalingEnabled: false,
+  } as CloudInfo;
+
+  let mapper: DeploymentParamsMapper;
+
+  mapper = new DeploymentParamsMapper(modelId, mlServerLimits, cloudInfo, true);
+
+  describe('DeploymentParamsMapper', () => {
+    describe('running in serverless', () => {
+      beforeEach(() => {
+        mapper = new DeploymentParamsMapper(
+          modelId,
+          {
+            max_single_ml_node_processors: 16,
+            total_ml_processors: 32,
+          },
+          {
+            isMlAutoscalingEnabled: false,
+          } as CloudInfo,
+          false
+        );
+      });
+
+      it('should enforce adaptive allocations', () => {
+        expect(
+          mapper.mapUiToUiDeploymentParams({
+            deploymentId: 'test-deployment',
+            optimized: 'optimizedForSearch',
+            adaptiveResources: false,
+            vCPUUsage: 'low',
+          })
+        ).toEqual({
+          adaptive_allocations: {
+            enabled: true,
+            max_number_of_allocations: 1,
+            min_number_of_allocations: 1,
+          },
+          deployment_id: 'test-deployment',
+          model_id: 'test-model',
+          priority: 'low',
+          threads_per_allocation: 1,
+        });
+      });
+    });
+
+    describe('32 cores, 16 single', () => {
+      beforeEach(() => {
+        mapper = new DeploymentParamsMapper(
+          modelId,
+          {
+            max_single_ml_node_processors: 16,
+            total_ml_processors: 32,
+          },
+          {
+            isMlAutoscalingEnabled: false,
+          } as CloudInfo,
+          true
+        );
+      });
+
+      it('should provide vCPU level', () => {
+        expect(mapper.getVCPURange('low')).toEqual({ max: 1, min: 1 });
+        expect(mapper.getVCPURange('medium')).toEqual({ max: 16, min: 2 });
+        expect(mapper.getVCPURange('high')).toEqual({ max: 32, min: 16 });
+      });
+    });
+
+    describe('when autoscaling in disabled', () => {
```

Comment:
Nit - guess this should read `is disabled` ?

Reply:
Fixed in c5f7e84cd69e880bca8c5aad5d345ff14d31c4fd